sbi/inference/snpe/snpe_base.py

```diff
@@ -202,6 +205,20 @@ def train(
         Returns:
             Density estimator that approximates the distribution $p(\theta|x)$.
         """
+        if self._round == 0 and self._neural_net is not None:
+            assert force_first_round_loss, (
+                "You have already trained this neural network. After you had trained "
+                "the network, you again appended simulations with `append_simulations"
+                "(theta, x)`, but you did not provide a proposal. If the new "
+                "simulations are sampled from the prior, you can set "
+                "`.train(..., force_first_round_loss=True`). However, if the new "
+                "simulations were not sampled from the prior, you should pass the "
+                "proposal, i.e. `append_simulations(theta, x, proposal)`. If "
+                "your samples are not sampled from the prior and you do not pass a "
+                "proposal and you set `force_first_round_loss=True`, the result of "
+                "SNPE will not be the true posterior. It will be systematically "
```

Comment:
it will be the 'proposal proposal posterior' which tends to be too narrow (reference? SNPEC paper?).